src/settings/BatchGroupConfigurationSettings/BatchGroupCongiurationForm.js

```diff
@@ -45,11 +188,28 @@ BatchGroupConfigurationForm.propTypes = {
   batchGroups: PropTypes.arrayOf(PropTypes.object),
   handleSubmit: PropTypes.func.isRequired,
   pristine: PropTypes.bool,
-  setSelectedBatchGroupId: PropTypes.func.isRequired,
+  selectBatchGroup: PropTypes.func.isRequired,
   submitting: PropTypes.bool,
+  form: PropTypes.object,
+  selectedBatchGroupId: PropTypes.string,
 };
 
 export default stripesFinalForm({
-  navigationCheck: true,
   subscription: { values: true },
+  navigationCheck: true,
+  mutators: {
+    setScheduledExport: ([value], state, tools) => {
+      if (!value) {
+        tools.changeValue(state, 'enableScheduledExport', () => false);
+        tools.changeValue(state, 'startTime', () => null);
+        tools.changeValue(state, 'weekdays', () => {});
+      } else {
+        tools.changeValue(state, 'enableScheduledExport', () => true);
+      }
+
+      if (value === SCHEDULE_EXPORT.daily) {
```

Comment:
As for me, it's weird to have `if {} else {}` and extra `if` in one code block. I would rather put it in something consistent - all `if`, without `elses`; or all in `if {} else if {} else`; or `switch`

Reply:
Fixed